protocol/src/errors.rs

```diff
@@ -55,6 +55,7 @@ error_chain! {
     foreign_links {
         Io(std::io::Error);
         FromUtf8(std::string::FromUtf8Error);
+        FromFfi(std::ffi::NulError);
```

Comment:
This variant isn't really from FFI itself, perhaps name this something more specific like `FromNulError`

Reply:
I didn't really get what is idiomatic here, fixed